### PR TITLE
smt2_parser: declare-datatypes for records

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 2be6d66c9976ca1444ab33a851d36ca7cde928b0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87881689

**Review by tautschnig — CHANGES_REQUESTED** [review]
This seems rather misleading to me: 1) the comment in the code says "process" but then the method doesn't actually do anything other than discarding tokens; 2) I failed to decipher the commit message.

File: src/solvers/smt2/smt2_parser.cpp [hunk 1]
```diff
@@ -1089,6 +1089,53 @@ typet smt2_parsert::function_signature_declaration()
   return mathematical_function_typet(domain, codomain);
 }
 
+/// process a datatype declaration (a Z3 extension)
```
Comment (martin-cs) [inline src/solvers/smt2/smt2_parser.cpp:1186]: Datatypes are in SMT-LIB; they aren't a Z3 extension any more.

**Review by martin-cs — COMMENTED** [review]
I think I'm with @tautschnig on this one; this just seems to ignore them.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 8cbdec49c1a460c1b0b477249d6d2a75bf387f42).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/92465976

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 5653817d708488c9867f61ea8acd94d47358ef6f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/93183039

**Comment by tautschnig** [comment]
The comment that @martin-cs made is still left to be addressed, otherwise this would be good to go.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 874ff04c648d7f51d75460e9ac2502b25fd931f5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/94083271

**Comment by TGWDB** [comment]
Closing this PR as no longer being worked on. Also linking to #6134  since the work here may be of interest there.

If you believe this is erroneous, please reopen and note why.

**Pull request closed without merging**